tinygrad/tensor.py

```diff
@@ -201,8 +201,11 @@ def shard_(self, devices:Tuple[str, ...], axis:Optional[int]=None):
   # ***** creation llop entrypoint *****
 
   @staticmethod
-  def _loadop(op, shape, device:Optional[str]=None, dtype:Optional[DType]=None, arg=None, **kwargs):
-    return Tensor(LazyBuffer.loadop(op, shape, dtype or dtypes.default_float, Device.canonicalize(device), arg), dtype=dtype, device=device, **kwargs)
+  def _loadop(op, shape, device:Optional[Union[Tuple[str], str]]=None, dtype:Optional[DType]=None, arg=None, **kwargs):
+    device, dtype, src = device if device else Device.canonicalize(device), dtype or dtypes.default_float, kwargs.get("src", None)
```

Comment:
don't you need to canonicalize each element if device is a tuple too?

also why do you need to get the src out? these creation function should not have src, and the single LazyBuffer case should remain the same

Reply:
Ye, you are right. Should be fixed now